libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -382,34 +382,53 @@ string IRGenerator::generateGetter(VariableDeclaration const& _varDecl)
 			{
 				if (returnTypes[i]->category() == Type::Category::Mapping)
 					continue;
-				if (auto arrayType = dynamic_cast<ArrayType const*>(returnTypes[i]))
+				auto const* arrayType = dynamic_cast<ArrayType const*>(returnTypes[i]);
+				if (arrayType)
 					if (!arrayType->isByteArray())
 						continue;
 
-				// TODO conversion from storage byte arrays is not yet implemented.
 				pair<u256, unsigned> const& offsets = structType->storageOffsetsOfMember(names[i]);
 				vector<string> retVars = IRVariable("ret_" + to_string(returnVariables.size()), *returnTypes[i]).stackSlots();
 				returnVariables += retVars;
 				code += Whiskers(R"(
 					<ret> := <readStorage>(add(slot, <slotOffset>))
 				)")
 				("ret", joinHumanReadable(retVars))
-				("readStorage", m_utils.readFromStorage(*returnTypes[i], offsets.second, true))
+				(
+					"readStorage",
+					arrayType ?
+						m_utils.copyArrayFromStorageToMemoryFunction(
+							dynamic_cast<ArrayType const&>(*arrayType->copyForLocation(DataLocation::Storage, arrayType->isPointer())),
```

Comment:
Can you set the pointer arguments to true/false? Does it matter?

Reply:
It doesn't meter and shouldn't make any difference in this case.
Also second one shouldn't be used with `copyForLocation` as it should already be correct type (it comes from `returnTypes`)

Will change this at both occurrences now